libs/muddle/src/rbc.cpp

```diff
@@ -534,6 +534,11 @@ void RBC::Deliver(SerialisedMessage const &msg, uint32_t sender_index)
   deliver_msg_callback_(miner_id, msg);
   lock_.lock();
```

Comment:
Here, unlocking/locking is done manually. Wouldn't it be safer to use RAII, like with `lock_guard` (but inverse)?

Reply:
yes, that would be better - I will make the change in another PR that I am working on currently

Reply:
May I suggest UnlockGuard for the name? :)

Reply:
sure :)